File: Manuel/개인 활동/Class 2/소수 찾기/find_prime.py
import sys

def is_prime(num):
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    
    for i in range(3, int(num ** 0.5) + 1, 2):
        if num % i == 0:
            return False
    return True

N = int(sys.stdin.readline())

numbers = list(map(int, sys.stdin.readline().split()))
prime_count = 0

for num in numbers:
    if is_prime(num):
        prime_count += 1

print(prime_count)

File: Manuel/개인 활동/Class 2/소수 찾기/find_prime.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

bool isPrime(int num)
{
    if(num < 2)
        return false;
    if(num == 2)
        return true;
    if(num % 2 == 0)
        return false;

    for(int i = 3; i <= sqrt(num); i += 2)
    {
        if(num % i == 0)
            return false;
    }
    return true;
}

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N;
    cin >> N;
    int prime_count = 0;

    for(int i = 0; i < N; i++)
    {
        int num;
        cin >> num;
        if(isPrime(num))
        {
            prime_count++;
        }
    }
    cout << prime_count << "\n";
    return 0;
}